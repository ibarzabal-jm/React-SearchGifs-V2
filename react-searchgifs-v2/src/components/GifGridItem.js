import React from 'react'

export const GifGridItem = ( {id, title, url} ) => {

    

    return (
        <div className="card animate__animated animate__rollIn">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

