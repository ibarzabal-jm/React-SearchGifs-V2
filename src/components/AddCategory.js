import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {
    
    const [inputValue, setInputValue] = useState(''); 
    
    const handleInputChange = ( e ) =>{
        setInputValue( e.target.value );

    }

    const handleSubmit = ( e ) => {
        
        e.preventDefault();

        if( inputValue.trim().length > 2){

            setCategories( categorias => [inputValue, ...categorias ] );
            setInputValue('');
        }else{
            
            console.log('Categor√≠as no agregada')
        }

    }

    return (
        <>
            <h4>Ingresa aqui abajo lo que quieras buscar</h4>
            <form onSubmit={handleSubmit}>
                <input
                    type = "text"
                    value = {inputValue}
                    onChange={ handleInputChange }
    
                />

            </form>
        </>
    )
}


AddCategory.propType = {
    setCategories: PropTypes.func.isRequired
}

