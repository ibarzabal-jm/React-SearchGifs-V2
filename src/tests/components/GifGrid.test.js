import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'

jest.mock('../../hooks/useFetchGifs')


describe('Prueba de componente <GifGrid />', () => {
    
    const category = 'Messi';

    test('debe de mostrar correctamente por pantalla ', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={category} /> );
        expect( wrapper ).toMatchSnapshot();
        
    })
    
    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{
            id:'ABC',
            title: 'Cualquiercosa',
            url:'https://localhost/juanma/cosa.jpg',
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={category} /> );
        
        expect( wrapper ).toMatchSnapshot();
    })
    
    
})
