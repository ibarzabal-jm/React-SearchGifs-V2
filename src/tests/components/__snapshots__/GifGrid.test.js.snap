// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de componente <GifGrid /> debe de mostrar correctamente por pantalla  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__shakeY"
  >
     
    Messi
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Prueba de componente <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__shakeY"
  >
     
    Messi
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquiercosa"
      url="https://localhost/juanma/cosa.jpg"
    />
  </div>
</Fragment>
`;
